tests/certs/CMakeLists.txt

```
@@ -0,0 +1,48 @@
+#***************************************************************************
+#                                  _   _ ____  _
+#  Project                     ___| | | |  _ \| |
+#                             / __| | | | |_) | |
+#                            | (__| |_| |  _ <| |___
+#                             \___|\___/|_| \_\_____|
+#
+# Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# This software is licensed as described in the file COPYING, which
+# you should have received as part of this distribution. The terms
+# are also available at https://curl.se/docs/copyright.html.
+#
+# You may opt to use, copy, modify, merge, publish, distribute and/or sell
+# copies of the Software, and permit persons to whom the Software is
+# furnished to do so, under the terms of the COPYING file.
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+# KIND, either express or implied.
+#
+# SPDX-License-Identifier: curl
+#
+###########################################################################
+transform_makefile_inc("Makefile.inc" "${CMAKE_CURRENT_BINARY_DIR}/Makefile.inc.cmake")
+include("${CMAKE_CURRENT_BINARY_DIR}/Makefile.inc.cmake")
+
+add_custom_target(clean-certs
+  COMMAND rm -f ${GENERATEDCERTS}
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_custom_target(build-certs
+  DEPENDS ${CERTCONFIGS} ${SRPFILES}
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genroot.sh EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh Server-localhost EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh Server-localhost.nn EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh Server-localhost0h EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh Server-localhost-firstSAN EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh Server-localhost-lastSAN EdelCurlRoot
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh stunnel EdelCurlRoot
+  COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/stunnel-sv.pem ${CMAKE_CURRENT_BINARY_DIR}/stunnel.pem
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/stunnel.pem
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/stunnel-sv.pem
+  COMMAND cp $< $@
```

```suggestion
add_custom_target(build-certs
  DEPENDS ${CERTCONFIGS} ${SRPFILES}
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genroot.sh" EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" Server-localhost EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" Server-localhost.nn EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" Server-localhost0h EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" Server-localhost-firstSAN EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" Server-localhost-lastSAN EdelCurlRoot
  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/genserv.sh" stunnel EdelCurlRoot
  COMMAND cp "${CMAKE_CURRENT_SOURCE_DIR}/stunnel-sv.pem" "${CMAKE_CURRENT_BINARY_DIR}/stunnel.pem"
  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
)

add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/stunnel.pem"
  DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/stunnel-sv.pem"
  COMMAND cp $< $@
```
quote paths with literal parts (style / syntax highlighting)

---

Updated